### PR TITLE
More detailed Pyrsia components and stories

### PR DESCRIPTION
**Pull request by erwin1**
This PR adds more detail to the architecture and use cases document. 
More info to be added, but I think it's more valuable to get this merged than to wait until it's complete.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [ ] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### CODE REVIEW

File: docs/developers/pyrsia-architecture-and-use-cases.md [hunk 1]
```diff
@@ -11,12 +11,18 @@
...
   verify transactions
 - Regular node: a Node using the network to download and validate software packages
-- Transaction: an operation in the transparency log. e.g. `add_artifact`,
-  `add_node`, ...
+- Transparency log: information about every operation on the Pyrsia network, like
+  adding or removing an artifact, but also about adding or removing authorized nodes.
+  A transparency log is stored as the payload of a blockchain transaction.
+- Blockchain : a distributed ledger including all the transactions in the Pyrsia
+  network
+- Block: a record of the distributed ledger including one ore more transactions
```
Comment (fishseabowl) [inline docs/developers/pyrsia-architecture-and-use-cases.md:19]:
```suggestion
- Block: a record of the distributed ledger including one or more transactions
```

File: docs/developers/pyrsia-architecture-and-use-cases.md [hunk 2]
```diff
@@ -109,38 +152,147 @@ artifacts and transparency logs.
...
+  this function is typically called from the CLI component when a build from
+  source was requested by a user.
+  - based on the package type it will:
+    - map the PSID to a source repo (if necessary)
+    - find a suitable build pipeline and trigger a build
+    - wait for the build result
+    - when the build finishes, hash the binary artifact and the source artifact
+      - one build might produce multiple (Pyrsia) artifacts
+      - for non-reproducible builds simple hashing will not be sufficient
+    - return the build result (could be multple: one per artifact) to the caller
```
Comment (fishseabowl) [inline docs/developers/pyrsia-architecture-and-use-cases.md:289]:
```suggestion
    - return the build result (could be multiple: one per artifact) to the caller
```

**Review by fishseabowl — APPROVED** [review]
LGTM

File: docs/developers/pyrsia-architecture-and-use-cases.md [hunk 3]
```diff
@@ -109,38 +152,147 @@ artifacts and transparency logs.
...
+
+### Verification service
+
+The build service is a component only used by authorized nodes. It implements all
+necessary logic to verify blockchain transactions.
+
+The verification service supports this function:
+
+- verify a transaction based on a its payload
+  this function is called from the blockchain component to verify a transaction
```
Comment (tiainen) [inline docs/developers/pyrsia-architecture-and-use-cases.md:254]: Perhaps rewrite this sentence to:

  - this function is called by the blockchain component to try and reach consensus on a transaction
    that was added by another authorized node by verifying the payload of that transaction.

**Review by tiainen — APPROVED** [review]

**Review by fishseabowl — APPROVED** [review]

**Pull request merged**